src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -390,6 +407,71 @@ function tryRemoveElement(elem) {
             });
         }
 
+        /**
+         * @private
+         */
+        onShakaError(error) {
+            console.error('shaka: error code', error.code, 'object', error);
+        }
+
+        /**
+         * @private
+         * @param e {Event} The event received from the `<video>` element
+         */
+        onShakaErrorEvent = (e) => {
+            // Extract the shaka.util.Error object from the event.
+            this.onShakaError(e.detail);
+        };
+
+        /* global Shaka */
+
+        /**
+         * @private
+         */
+        setSrcWithShakaPlayer(elem, options, url) {
+            requireShakaPlayer().then(() => {
+                Shaka.polyfill.installAll();
+                if (Shaka.Player.isBrowserSupported()) {
+                    const shakaPlayer = new Shaka.Player(elem);
+
+                    shakaPlayer.configure({
+                        streaming: {
+                            retryParameters: {
+                                maxAttempts: 6
+                            },
+                            forceTransmuxTS: true,
+                            rebufferingGoal: 5,
+                            bufferingGoal: 30,
+                            bufferBehind: 30,
+                            inaccurateManifestTolerance: 5
+                        },
+                        abr: {
+                            enabled: false
+                        }
+                    });
+
+                    shakaPlayer.addEventListener('error', this.onShakaErrorEvent);
+
+                    shakaPlayer.load(url).then(() => {
+                        console.debug('shaka: loaded manifest');
+                    }).catch((err) => {
+                        this.onShakaError(err);
+                        return Promise.reject();
+                    });
+
+                    this._shakaPlayer = shakaPlayer;
+
+                    // This is needed in setCurrentTrackElement
+                    this.#currentSrc = url;
+
+                    return Promise.resolve();
```

Comment:
I think we should wait for Shaka to load `url`:
```suggestion
                    this._shakaPlayer = shakaPlayer;

                    shakaPlayer.load(url).then(() => {
                        console.debug('shaka: loaded manifest');

                        // This is needed in setCurrentTrackElement
                        this.#currentSrc = url;

                        return Promise.resolve();
                    }).catch((err) => {
                        this.onShakaError(err);
                        return Promise.reject();
                    });
```